src/api.c

```diff
@@ -1321,6 +1321,60 @@ static void cgroup_free_cg_mount_table(void)
 	memset(&cg_cgroup_v2_empty_mount_paths, 0, sizeof(cg_cgroup_v2_empty_mount_paths));
 }
 
+/*
+ * Parses the mount options of the given mount point and checks for the
+ * option in the list of mount options.
+ * @mnt: Mount point name to search for mount points.
+ * @mnt_option: Mount option to be searched.
+ *
+ * Returns 0, in case of success and ECGOTHER on failure.
+ */
+static int check_mount_point_opt(char *mnt, char *mnt_option)
+{
+	struct mntent *ent, *temp_ent = NULL;
+	char mntent_buffer[4 * FILENAME_MAX];
+	char *mntopt = NULL;
+	FILE *proc_mount;
+	int ret = 0;
+
+	proc_mount = setmntent(mnt, "r");
+	if (!proc_mount) {
+		cgroup_err("cannot open %s: %s\n", mnt, strerror(errno));
+		last_errno = errno;
+		ret = ECGOTHER;
+		goto err;
+	}
+
+	temp_ent = (struct mntent *) malloc(sizeof(struct mntent));
+	if (!temp_ent) {
+		last_errno = errno;
+		ret = ECGOTHER;
+		goto err;
+	}
+
+	ent = getmntent_r(proc_mount, temp_ent,	mntent_buffer, sizeof(mntent_buffer));
+	if (!ent) {
+		last_errno = errno;
+		ret = ECGOTHER;
+		goto err;
+	}
+
+	mntopt = hasmntopt(ent, mnt_option);
```

Comment:
This [commit](a18ae861e77f106517df203e30a810ba59b082d9) added smarts to ensure that `hasmntopt()` matched the whole opt string and not just a substring.  Do we want to do that here?

Reply:
Great suggestion, yeah we might need something like that.